src/sentry/integrations/vercel/uninstall.py

```diff
@@ -1,17 +1,104 @@
+import hashlib
+import hmac
 import logging
 
+from django.utils.crypto import constant_time_compare
 from django.views.decorators.csrf import csrf_exempt
+from requests.exceptions import RequestException
 
+from sentry import VERSION, http, options
 from sentry.api.base import Endpoint
-from sentry.models import AuditLogEntryEvent, Integration, Organization, OrganizationIntegration
+from sentry.models import (
+    AuditLogEntryEvent,
+    Integration,
+    Organization,
+    OrganizationIntegration,
+    Project,
+    SentryAppInstallationForProvider,
+    SentryAppInstallationToken,
+)
+from sentry.shared_integrations.exceptions import IntegrationError
 from sentry.utils.audit import create_audit_entry
+from sentry.utils.compat import filter
+from sentry.utils.http import absolute_uri
 
 logger = logging.getLogger("sentry.integrations.vercel.uninstall")
 
 
+class NoCommitFoundError(IntegrationError):
+    pass
+
+
+def verify_signature(request):
+    signature = request.META["HTTP_X_VERCEL_SIGNATURE"]
+    secret = options.get("vercel.client-secret")
+
+    expected = hmac.new(
+        key=secret.encode("utf-8"), msg=bytes(request.body), digestmod=hashlib.sha1
+    ).hexdigest()
+    return constant_time_compare(expected, signature)
+
+
+def safe_json_parse(resp):
+    if resp.headers.get("content-type") == "application/json":
+        return resp.json()
+    return None
+
+
+def get_payload_and_token(payload, organization_id, sentry_project_id):
```

Comment:
@MeredithAnya why are all these functions in `uninstall.py`? Why not `webhook.py`?

Reply:
@scefali I'm going to add more info to the description later but https://github.com/getsentry/sentry/pull/24322 has some context. the delete hook became the generic hook, and the creation of webhooks per install are going to be deprecated.